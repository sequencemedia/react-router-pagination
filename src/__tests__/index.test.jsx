import React from 'react'

import {
  toSnapshot
} from 'react-component-snapshot'

import '@testing-library/jest-dom'

import {
  render
} from '@testing-library/react'

import {
  MemoryRouter
} from 'react-router'

import Pagination from '#pagination'

describe('#pagination', () => {
  describe('<Pagination />', () => {
    describe('With `pageNumber` and `totalPages`', () => {
      it('renders', () => {
        expect(toSnapshot(render(
          <MemoryRouter>
            <Pagination
              pageNumber={1}
              totalPages={2}
            />
          </MemoryRouter>
        )))
          .toMatchSnapshot()
      })
    })

    describe('With `pageNumber` and `totalPages` and `spread`', () => {
      it('renders', () => {
        expect(toSnapshot(render(
          <MemoryRouter>
            <Pagination
              pageNumber={9}
              totalPages={9}
              spread={3}
            />
          </MemoryRouter>
        )))
          .toMatchSnapshot()
      })
    })
  })

  describe('<Pagination format=\'center\' />', () => {
    describe('With `pageNumber` and `totalPages`', () => {
      it('renders', () => {
        expect(toSnapshot(render(
          <MemoryRouter>
            <Pagination
              format='center'
              pageNumber={1}
              totalPages={2}
            />
          </MemoryRouter>
        )))
          .toMatchSnapshot()
      })
    })

    describe('With `pageNumber` and `totalPages` and `spread`', () => {
      it('renders', () => {
        expect(toSnapshot(render(
          <MemoryRouter>
            <Pagination
              format='center'
              pageNumber={9}
              totalPages={9}
              spread={3}
            />
          </MemoryRouter>
        )))
          .toMatchSnapshot()
      })
    })
  })
})
