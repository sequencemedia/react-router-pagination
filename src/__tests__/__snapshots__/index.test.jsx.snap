// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#pagination <Pagination /> With \`pageNumber\` and \`totalPages\` and \`spread\` renders 1`] = `
<ul
  className="pagination"
>
  <li
    className="zero-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/1"
        onClick={[Function]}
      >
        <span
          className="page-number"
        >
          1
        </span>
      </a>
    </Link>
  </li>
  <li
    className="reverse-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/6"
        onClick={[Function]}
      >
        <span
          className="reverse"
        >
          ‹
        </span>
      </a>
    </Link>
  </li>
  <Fragment>
    <li
      className="page"
    >
      <Link>
        <a
          data-discover="true"
          href="/7"
          onClick={[Function]}
        >
          <span
            className="page-number"
          >
            7
          </span>
        </a>
      </Link>
    </li>
    <li
      className="page"
    >
      <Link>
        <a
          data-discover="true"
          href="/8"
          onClick={[Function]}
        >
          <span
            className="page-number"
          >
            8
          </span>
        </a>
      </Link>
    </li>
    <li
      className="current-page"
    >
      <Link>
        <a
          data-discover="true"
          href="/9"
          onClick={[Function]}
        >
          <span
            className="page-number"
          >
            9
          </span>
        </a>
      </Link>
    </li>
  </Fragment>
</ul>
`;

exports[`#pagination <Pagination /> With \`pageNumber\` and \`totalPages\` renders 1`] = `
<ul
  className="pagination"
>
  <Fragment>
    <li
      className="current-page"
    >
      <Link>
        <a
          data-discover="true"
          href="/1"
          onClick={[Function]}
        >
          <span
            className="page-number"
          >
            1
          </span>
        </a>
      </Link>
    </li>
  </Fragment>
  <li
    className="last-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/2"
        onClick={[Function]}
      >
        <span
          className="page-number"
        >
          2
        </span>
      </a>
    </Link>
  </li>
</ul>
`;

exports[`#pagination <Pagination format='center' /> With \`pageNumber\` and \`totalPages\` and \`spread\` renders 1`] = `
<ul
  className="pagination"
>
  <li
    className="zero-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/1"
        onClick={[Function]}
      >
        <span
          className="page-number"
        >
          1
        </span>
      </a>
    </Link>
  </li>
  <li
    className="reverse-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/6"
        onClick={[Function]}
      >
        <span
          className="reverse"
        >
          ‹
        </span>
      </a>
    </Link>
  </li>
  <Fragment>
    <li
      className="page"
    >
      <Link>
        <a
          data-discover="true"
          href="/7"
          onClick={[Function]}
        >
          <span
            className="page-number"
          >
            7
          </span>
        </a>
      </Link>
    </li>
    <li
      className="page"
    >
      <Link>
        <a
          data-discover="true"
          href="/8"
          onClick={[Function]}
        >
          <span
            className="page-number"
          >
            8
          </span>
        </a>
      </Link>
    </li>
    <li
      className="current-page"
    >
      <Link>
        <a
          data-discover="true"
          href="/9"
          onClick={[Function]}
        >
          <span
            className="page-number"
          >
            9
          </span>
        </a>
      </Link>
    </li>
  </Fragment>
</ul>
`;

exports[`#pagination <Pagination format='center' /> With \`pageNumber\` and \`totalPages\` renders 1`] = `
<ul
  className="pagination"
>
  <li
    className="current-page"
  >
    <Link>
      <a
        data-discover="true"
        href="/1"
        onClick={[Function]}
      >
        <span
          className="page-number"
        >
          1
        </span>
      </a>
    </Link>
  </li>
  <li
    className="page"
  >
    <Link>
      <a
        data-discover="true"
        href="/2"
        onClick={[Function]}
      >
        <span
          className="page-number"
        >
          2
        </span>
      </a>
    </Link>
  </li>
</ul>
`;
